File: ms-book-springboot-rest-hibernate/src/main/java/com/performance/book/model/Book.java
package com.performance.book.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "book")
public class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    private String isbn;
    private Long languageId;
    private Integer numPages;
    private LocalDate publicationDate;
    private Long publisherId;
}

File: ms-book-go-mux-gorm/entity/book.go
package entity

import (
	"ms-book-go-mux-gorm/types"
)

type Book struct {
	ID              int        `json:"id"`
	Title           string     `json:"title"`
	Isbn            string     `json:"isbn"`
	PublicationDate types.Date `json:"publication_date"`
	LanguageId      int        `json:"language_id"`
	NumPages        int        `json:"num_pages"`
	PublisherId     int        `json:"publisher_id"`
}

type Tabler interface {
	TableName() string
}

// TableName overrides the table name used by User to `profiles`
func (Book) TableName() string {
	return "book"
}
